src/main/java/com/verde/market/persistence/ProductoRepository.java

```diff
@@ -0,0 +1,57 @@
+package com.verde.market.persistence;
+
+import com.verde.market.domain.Product;
+import com.verde.market.domain.repository.ProductRepository;
+import com.verde.market.persistence.crud.ProductoCrudRepository;
+import com.verde.market.persistence.entity.Producto;
+import com.verde.market.persistence.mapper.ProductMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public class ProductoRepository implements ProductRepository {
+
```

Comment:
Same comment regarding DI. Is easier to test if you use DI base on setter or constructor.